### PR TITLE
Canonicalization of union array should propagate required only if defined

### PR DESCRIPTION
**Pull request by trevorr**

### CODE REVIEW

File: src/canonical.js
Comment (ForsakenHarmony) [inline src/canonical.js]: Could you change this to `typeof form.required === 'boolean'`?

**Review by ForsakenHarmony — CHANGES_REQUESTED** [review]

**Review by ForsakenHarmony — APPROVED** [review]

**Comment by trevorr** [comment]
Is this waiting for another reviewer?

File: src/canonical.js
Comment (jstoiko) [inline src/canonical.js:71]: in what context would items have `required` set here? i.e. do we even need this line?

Comment (trevorr) [reply src/canonical.js:71]: There's a test failure if it is removed:

```
  1) canonicalForm() should generate canonical form of type Songs.Musician:

      AssertionError: expected { Object (type, additionalProperties, ...) } to deeply equal { Object (type, additionalProperties, ...) }
      + expected - actual

                     }
                   }
                   "type": "object"
                 }
      -          "required": [undefined]
      +          "required": true
                 "type": "array"
               }
               "name": {
                 "required": true
                     }
                   }
                   "type": "object"
                 }
      -          "required": [undefined]
      +          "required": true
                 "type": "array"
               }
               "name": {
                 "required": true
```

While it seems strange to set `required` on `items`, I think this must be used to propagate `required` when hoisting arrays of unions (to form a union of arrays).

Comment (jstoiko) [reply src/canonical.js:71]: that makes sense. then let's keep it. will merge & release this today.

**Comment by jstoiko** [comment]
will merge this and make the release once #79 is figured-out.

thanks again x1000 @trevorr !

**Comment by trevorr** [comment]
@jstoiko Thanks for getting that merged and released so quickly. I do have one more (hopefully last) fix to submit, but I'm still working on getting a small reproducing test case and root-causing the issue. Basically, I'm seeing a union get `properties: {}` added to it, which is making `findClass` return `object` instead of `union`. The simple fix is just trusting `form.type` if it's defined (and then falling back to seeing if `properties` is defined), but I'm still trying to figure out where the extraneous `properties: {}` is coming from.

**Pull request merged**
